File: c++/leetcode/941. Valid Mountain Array.cpp
class Solution
{
public:
  bool validMountainArray(vector<int> &A)
  {
    const int sz = A.size();
    if (sz < 3)
      return false;

    int left = 0;

    // walk to peak
    while (A[left] < A[left + 1])
    {
      ++left;
      if (left == sz - 1)
        return false;
    }

    if (left < 1)
      return false;

    // walk down peak
    while (left < sz - 1)
    {
      if (A[left] <= A[left + 1])
        return false;
      ++left;
    }

    return true;
  }
};

File: javascript/leetcode/941. Valid Mountain Array.js
/**
 * @param {number[]} A
 * @return {boolean}
 */
var validMountainArray = function (A) {
  const sz = A.length;
  if (sz < 3) return false;

  let left = 0;

  while (A[left] < A[left + 1]) ++left;
  if (left === sz - 1) return false;

  if (left < 1) return false;

  while (left < sz - 1) {
    if (A[left] <= A[left + 1]) return false;
    ++left;
  }

  return true;
};
